rust/stackablectl/src/cmds/release.rs

```diff
@@ -319,6 +356,87 @@ async fn install_cmd(
     }
 }
 
+#[instrument(skip_all, fields(indicatif.pb_show = true))]
+async fn upgrade_cmd(
+    args: &ReleaseUpgradeArgs,
+    cli: &Cli,
+    release_list: release::ReleaseList,
+    transfer_client: &xfer::Client,
+) -> Result<String, CmdError> {
+    info!(release = %args.release, "Upgrading release");
+    Span::current().pb_set_message("Upgrading release");
+
+    match release_list.get(&args.release) {
+        Some(release) => {
+            let mut output = cli.result();
+            let client = Client::new().await.context(KubeClientCreateSnafu)?;
+
+            // Get all currently installed operators to only upgrade those
+            let installed_charts: Vec<Release> =
+                helm::list_releases(&args.operator_namespace).context(HelmSnafu)?;
+
+            let mut operators: Vec<String> = operator::VALID_OPERATORS
+                .iter()
+                .filter(|operator| {
+                    installed_charts
+                        .iter()
+                        .any(|release| release.name == utils::operator_chart_name(operator))
+                })
+                .map(|operator| operator.to_string())
+                .collect();
+
+            // Uninstall the old operator release first
+            release
+                .uninstall(
+                    &operators,
+                    &args.excluded_products,
+                    &args.operator_namespace,
+                )
+                .context(ReleaseUninstallSnafu)?;
+
+            // If operators were added to args.included_products, install them as well
+            for product in &args.included_products {
+                if !operators.contains(product) {
+                    operators.push(product.clone());
+                }
+            }
+
+            // Upgrade the CRDs for all the operators to be upgraded
+            release
+                .upgrade_crds(
+                    &operators,
+                    &args.excluded_products,
+                    &args.operator_namespace,
+                    &client,
+                    transfer_client,
+                )
+                .await
+                .context(CrdUpgradeSnafu)?;
+
+            // Install the new operator release
+            release
+                .install(
+                    &operators,
+                    &args.excluded_products,
+                    &args.operator_namespace,
+                    &ChartSourceType::from(cli.chart_type()),
+                )
+                .await
+                .context(ReleaseInstallSnafu)?;
+
+            output
+                .with_command_hint(
+                    "stackablectl operator installed",
+                    "list installed operators",
+                )
+                .with_output(format!("Upgraded to release '{}'", args.release));
+
+            Ok(output.render())
+        }
+        None => Ok("No such release".into()),
```

Comment:
I think this is supposed to be an Err, and maybe we can say what the invalid release is:

Something like:

```suggestion
        None => Err(NoSuchReleaseSnafu { release: &args.release }),
```

Reply:
https://github.com/stackabletech/stackable-cockpit/pull/379/commits/b4003986e817e0063e997f7dc080c82f7e9f0648